Methane_MarkDown.Rmd

```diff
@@ -347,14 +346,140 @@ expsen_data<- sensitivity_emission%>%
   geom_line(aes(y= Mean_sen, color= variable), size= 0.5)+
   facet_wrap(~variable, scales = "free_y")+
   labs( title= "Multi Parameter vs Natural CH4 Run", y= "Mean")
+```
+
+
+## Relative Importance
+
+Relative importance is used to understand how great of an impact different parameters have on a variable. Since our parameters impact one another as well, we must look at there impacts on the variable relative to each another. To do this we first alter a data frame to exclude information that may prevent our code from working.
+
+```{r rel-data, message= FALSE, warning=FALSE, results= FALSE}
+rell_data<- full_data %>% 
+  dplyr::select(-scenario, -units, -run_number)
+```
+
+Then with R package `relaimpo`, we create a function that filters by year and variable, and then runs a relative importance test (`calc.relimp`).
+
+```{r rel-relaimpo, include=FALSE}
+library(relaimpo)
+```
+
+
+```{r rel-func, message= FALSE, warning=FALSE, results= FALSE}
+relative_function<- function(yearask, var) {
+  output_year<- rell_data %>%
+  filter(year== yearask, variable== var) %>%
+  dplyr::select(LIFETIME_SOIL, LIFETIME_STRAT, NATURAL_CH4, PREINDUSTRIAL_CH4, value)
+  output_year
+  
+lm_data<- lm(value ~., data= output_year )
+summary(lm_data)
+lm_data
+
+result<- calc.relimp(lm_data, rela= TRUE)
+output<- result$lmg
+output$year<- yearask
+output$variable<- var
+output
+}
+```
+
+Using `lapply`, we can run our function with a selected variable and years (in an `as.list`). This makes our function run as many times as there are years asked for. We then `bind_rows` into a single data frame and repeat for all parameters.
+
+```{r rel-lapply, message= FALSE, warning=FALSE, results= FALSE}
+lap_Co2Concentration<- lapply(as.list(2000:2100), relative_function, "CO2_concentration")
+Rel_Co2Concentration<- bind_rows(lap_Co2Concentration)
+
+lap_RF_tot<- lapply(as.list(2000:2100), relative_function, "RF_tot")
+Rel_RF_tot<- bind_rows(lap_RF_tot)
+
+lap_RF_CO2<- lapply(as.list(2000:2100), relative_function, "RF_CO2")
+Rel_RF_CO2<- bind_rows(lap_RF_CO2)
+
+lap_global_tas<- lapply(as.list(2000:2100), relative_function, "global_tas")
+Rel_global_tas<- bind_rows(lap_global_tas)
+
+```
+
+Visualizing this data with stacked area graphs can make it more apparent the trend of relative importance of each variable.
+
+```{r rel-graphs, echo= FALSE}
+#Co2 Concentration
+Rel_Co2Concentrationlong<- Rel_Co2Concentration %>%
+  pivot_longer(!c(year, variable), names_to = "parameter", values_to = "value")
+
+Rel_cconc_graph<-
+  ggplot(Rel_Co2Concentrationlong, aes(x= year, y= value, fill= parameter))+
+  geom_area()+
+  labs(title= "Relitive Importance Co2 Concentration")
+
+#RF Total
+Rel_RF_totlong<- Rel_RF_tot %>%
+  pivot_longer(!c(year, variable), names_to = "parameter", values_to = "value")
+
+Rel_RF_graph<- 
+  ggplot(Rel_RF_totlong, aes(x= year, y= value, fill= parameter))+
+  geom_area()+
+  labs(title= "Relitive Importance RF Total")
```

Comment:
Here and elsewhere: "Relative"

Reply:
<img width="686" alt="image" src="https://user-images.githubusercontent.com/106257875/177409523-03e0557e-109f-4b98-9c4e-4cfc40e906de.png">

Reply:
<img width="659" alt="image" src="https://user-images.githubusercontent.com/106257875/177409666-00a9b5fa-c3fb-4016-8341-2a5d3be63b15.png">

Reply:
<img width="681" alt="image" src="https://user-images.githubusercontent.com/106257875/177409795-caa667a2-17af-496d-b254-c231606efb57.png">

Reply:
Niiiiiice!